src/meta/app/src/principal/user_stage.rs

```diff
@@ -56,6 +56,9 @@ use crate::storage::StorageParams;
 // ON_ERROR = { CONTINUE | SKIP_FILE | SKIP_FILE_<num> | SKIP_FILE_<num>% | ABORT_STATEMENT }
 // SIZE_LIMIT = <num>
 
+pub const COPIED_FILES_MAX_COMMIT_NUM: usize = 15000;
+pub const COPY_WITH_MAX_FILE_MSG: &str = "copy into table can commit at most 15000 paths once. Please use CopyOption 'max_files=<num>' to limit the number of files and run multi times until result is empty or use 'force=true, purge=true' to avoid committing paths if acceptable.";
```

Comment:
Only need `purge=true`?

Reply:
we used to require `force=true, purge=true`. 
https://github.com/datafuselabs/databend/pull/11981

do you mean filter but not commit when `purge=true ` only.   also makes sense.
I think this is a feature that we can define. better if user have more choices?

`purge=true` is not reliable since delete may fail (and we do not rollback in this case not)?

Reply:
`force=true, purge=true` this is not a security way let the user to do, if the purge failed, next time the copy may double the datas.

Reply:
what about add another  options `commit_paths = true|false`, which is easier to understand?